bleep-core/src/main/scala/bleep/BuildPaths.scala

```diff
@@ -49,8 +33,28 @@ object BuildPaths {
     case object BSP extends Mode
   }
 
-  def fromBleepJson(cwd: Path, bleepJsonFile: Path, mode: Mode): BuildPaths =
-    fromBuildDir(cwd, bleepJsonFile.getParent, mode)
+  def find(cwd: Path, mode: Mode): Either[BuildException.BuildNotFound, BuildPaths] = {
+    // keep looking up until we find build file
+    def findBleepJson(from: Path): Option[Path] = {
+      def is(f: File): Option[File] =
+        Option(f.list()) match {
```

Comment:
instead of listing, why not just check if if known file exists?
new File(f, constants.BuildFileName).exists

is probably a bit faster

Reply:
Yeah, that should be faster, I'll fix it

BTW if you have any cycles to review more I'd be happy to break this up into more digestible PRs.